src/EditorFeatures/TestUtilities/LanguageServer/AbstractLanguageServerProtocolTests.cs

```diff
@@ -227,15 +228,15 @@ private static string GetDocumentFilePathFromName(string documentName)
         /// Adds/Removes the document instead of updating file path due to
         /// https://github.com/dotnet/roslyn/issues/34837
         /// </summary>
-        private static Solution ChangeDocumentFilePathToValidURI(Solution originalSolution, TestHostDocument originalDocument, SourceText text)
+        private static Solution ChangeDocumentFilePathToValidURI(Solution solution, TestHostDocument document, SourceText text)
         {
-            var documentName = originalDocument.Name;
+            var documentId = document.Id;
+            var documentName = document.Name;
             var documentPath = GetDocumentFilePathFromName(documentName);
 
-            var solution = originalSolution.RemoveDocument(originalDocument.Id);
-
-            var newDocumentId = DocumentId.CreateNewId(originalDocument.Project.Id);
-            return solution.AddDocument(newDocumentId, documentName, text, filePath: documentPath);
+            return solution
```

Comment:
It's important to preserve the document id since TestWorkspace uses it to map between TestHostDocument and Document.

Reply:
Ahh interesting.  Good to know